invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/index.js

```diff
@@ -96,3 +97,41 @@ if (sidebarCommunitiesManageDiv) {
     sidebarCommunitiesManageDiv
   );
 }
+
+const requestAccessButton = document.getElementById("request-access-button");
+const handler = async () => {
+  // TODO i have a feeling this could be imported from somewhere else
+  const apiConfig = {
+    withCredentials: true,
+    xsrfCookieName: "csrftoken",
+    xsrfHeaderName: "X-CSRFToken",
+  };
+
+  const axiosWithConfig = axios.create(apiConfig);
+  const recid = requestAccessButton.dataset.recid;
+  const result = await axiosWithConfig.post(`/api/records/${recid}/access/request`, {
+    headers: {
+      Accept: "application/json",
+    },
+  });
+
+  const status = result.status;
+  if(status == 200) {
+    const id = result.data.id;
+
+    // TODO it would be nicer to have the backend report the proper `links.self_html`
+    //      instead of constructing the URL here
+    const url = `/me/requests/${id}`
+
+    // redirection here because opening in new tab might be blocked as popup
+    window.location.href = url;
+
+  } else {
+    // TODO more proper error handling, based on return code & response data
+    alert(`Something went wrong while requesting access (code ${status}): ${result.data}`);
+  }
+};
+
+if (requestAccessButton) {
+  requestAccessButton.addEventListener("click", handler);
```

Comment:
this should be a react component with state management, I think. 
We could have a quick chat about it if you prefer

Reply:
for inspiration, very similar button:
https://github.com/zenodo/zenodo-rdm/pull/354/files#diff-e1367c1098cbefb69016855a6b19a167c3afdcdeb4c4ca04edfaa32fdcc69157R14

Reply:
the entire access request form is now a react component :)